### Taller 2

2. Suponga que una moneda justa es lanzada n veces, donde n es un numero par, y x representa
el numero de caras. Determine el numero esperado de caras y la probabilidad que el numero
de caras es exactamente x =
n
2
, asumiendo n = 2, 4, 8, 16, 32, 64, 128. Explique los resultados.

In [7]:
def trianguloPascal(n):
 
    # creamos una lista que contendra los dos primeras lineas
    lista = [[1],[1,1]]
 
    # bucle que se generara tantas veces como lineas vayamos a tener
    for i in range(1,n):
 
        # inicializamos la linea
        linea = [1]
 
        # bucle por cada uno de los valores de la anterior linea
        for j in range(0,len(lista[i])-1):
 
            # añadimos a la lista los nuevos valores
            # sumamos el valor de la lista anterior con el siguinte
            linea.extend([ lista[i][j] + lista[i][j+1] ])
 
        # añadimos el ultimo valor a la nueva linea
        # siempre es un 1 igual que el primero
        linea += [1]
 
        # añadimos la linea a la lista
        lista.append(linea)
 
    #devolvemos la lista ya creada
    return lista
# ref: https://www.lawebdelprogramador.com/codigo/Python/2617-Triangulo-de-Pascal.html

In [8]:
def expected_value(probabilitys):
    E = 0
    for i in range(len(probabilitys)):
        E += i * probabilitys[i]
    return E

In [9]:
n_vector = [2, 4, 8, 16, 32, 64, 128]
pascal_n = trianguloPascal(n_vector[len(n_vector) -1])
E = []
for n in n_vector:
    probabilitys = []
    cases_posibility = 2**n
    for i in range(n + 1):
        probability = pascal_n[n][i]/cases_posibility
        if(i == n / 2):
            print("x=%s la probabilidad es %s"%(i,probability))
        probabilitys.append(probability)
        #print("%s  | %s"%(i,probability))
    E.append(expected_value(probabilitys))
for i in range(len(n_vector)):
    print("el valor esperado de caras con n=%s es: %s "%(n_vector[i],E[i]))

x=1 la probabilidad es 0.5
x=2 la probabilidad es 0.375
x=4 la probabilidad es 0.2734375
x=8 la probabilidad es 0.196380615234375
x=16 la probabilidad es 0.13994993409141898
x=32 la probabilidad es 0.09934675374796689
x=64 la probabilidad es 0.07038609217001514
el valor esperado de caras con n=2 es: 1.0 
el valor esperado de caras con n=4 es: 2.0 
el valor esperado de caras con n=8 es: 4.0 
el valor esperado de caras con n=16 es: 8.0 
el valor esperado de caras con n=32 es: 16.0 
el valor esperado de caras con n=64 es: 31.999999999999996 
el valor esperado de caras con n=128 es: 63.99999999999997 


n/2 es el valor esperado de las caras si n es par

3. Suponga que el numero de errores en un programa de computador tiene una distribuci´on de
Poisson con par´ametro λ = 5.
(a) Determine la probabilidad que el programa de computador no tenga errores.
(b) La probabilidad que el n´umero de errores son 5 o m´as.


In [10]:
import math
def poison_function(lambda_poisson, k_poisson):
    return ((lambda_poisson**k_poisson)*math.e**(-lambda_poisson))/(math.factorial(k_poisson))

In [11]:
k=0
l_poisson = 5
print(poison_function(l_poisson,k))

0.006737946999085469


La probabilidad de que no ocurran errores es de 0.67 %

In [12]:
k_poisson= [0, 1, 2, 3, 4]
l_poisson = 5
probability = 0
for k in k_poisson:
    probability += poison_function(l_poisson,k)
probality_5 = 1 -probability
print(probality_5)

0.5595067149347875


b) La probabilidad que el n´umero de errores son 5 o mas es : 0.5595067149347875